### DISCUSSION TITLE
Notification trigger for analysis dryrun status

### DISCUSSION THREAD

**Discussion by ilyastoliar** [argoproj/argo-rollouts#2129] [Q&A]
Hi,
Is there any possible method to trigger notification based on analysis dryrun status?
I have tried to define it with [custom triggers](https://argoproj.github.io/argo-rollouts/features/notifications/#custom-triggers), but with no success.
All i get is `failed to execute when condition: cannot fetch status from <nil> (1:21)\n | analysisrun.status.dryrunsummary.count != 2` .
I have tired to add the notification annotation both to the Rollout and to the AnalysisTemplate.

**Comment by RaviHari** [comment]
Can you please share the details:
1. argo-rollout version used.
2. configmap used

**Reply by ilyastoliar** [reply]
Version: `kubectl-argo-rollouts: v1.2.0+08cf10e`
configmap:

``` 
apiVersion: v1
kind: ConfigMap
metadata:
  name: argo-rollouts-notification-configmap
  namespace: argo-rollouts

data:
  service.slack: |
    token: $slack-token
  template.dryrun: |
    message: |
      Rollout {{.rollout.metadata.name}} dryrun
    slack:
        attachments: |
            [{
              "title": "{{ .rollout.metadata.name}}",
              "color": "#800080"
            }]
 
   trigger.on-dryrun: |
      - send: [dryrun]
        when: analysisrun.status.dryrunsummary.count != '2'

**Reply by ilyastoliar** [reply]
@RaviHari :)

**Comment by ilyastoliar** [comment]
@RaviHari any luck?

**Comment by agrawroh** [comment]
@ilyastoliar This seems quite useful. `AnalysisRun` is an instantiation of `AnalysisTemplate` and my hunch is that the notification you are adding to the `AnalysisTemplate` is not being copied over to the `AnalysisRun`. Also, I think having the notification on `AnalysisTemplate` is probably not useful in this case as it'd fail the trigger spec (.status.dryRunSummary is not part of the **AnalysisTemplate** but **AnalysisRun**).

@RaviHari should we be passing on the annotations to the `AnalysisRun` from `AnalysisTemplate` while instantiating it? Also, is there a way to make sure that those annotations which are applicable only to `AnalysisRun` should be removed/not executed for `AnalysisTemplate` or otherwise they'd always fail.
